.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: Apache-2.0
+name: "CodeQL"
+
+on:
+  push:
+    branches: [master, release-1.0, release-1.1, release-1.2, release-1.3, release-1.4, release-2.0, release-2.1, release-2.2]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [master]
+  schedule:
+    - cron: '0 5 * * 6'
+
+jobs:
+  analyse:
+    name: Analyse
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        # We must fetch at least the immediate parents so that if this is
+        # a pull request then we can checkout the head.
+        fetch-depth: 2
+
+    # If this run was triggered by a pull request event, then checkout
+    # the head of the pull request instead of the merge commit.
+    - run: git checkout HEAD^2
```

Comment:
Why checkout prior commit? Don't we want to analyze the PR commit?

What if the PR has multiple commits, HEAD^2 wouldn't be correct in that case...

Reply:
The merge would be HEAD, HEAD~1 should be the one before that - @btl5037 ?

Reply:
```Run git checkout HEAD^2
Previous HEAD position was c36d1bf Merge 6739eb0013f82e8be3346daf78fbcff3c3b88a22 into af0d871b3b3cf9915ed023c0d013b1ef0721ef12
HEAD is now at 6739eb0 Create codeql-analysis.yml```

looks right to me